### ISSUE TITLE
Ensure code review with integrating PullApprove 

### ISSUE THREAD

**Issue by SecUpwN** [SecUpwN/Android-IMSI-Catcher-Detector#708]
[![PullApprove](https://pullapprove.com/static/img/logos/pull-approve-logo-gray-dk.png)](https://pullapprove.com/)

I've just been crawling through [GitHub Integrations](https://github.com/integrations/) and found ![PullApprove](https://github.com/integrations/pullapprove). Since we've too often had the case that stuff gets merged and later on some developers like @agilob complained that noone seemed to have reviewed it before the merge, `PullApprove` is an option to prevent this in the future by *requiring* an approval statement as simple as "approved" or "lgtm" by the previously defined developers. Means:

[![Workflow](https://pullapprove.com/static/img/flow-diagram.png)](https://pullapprove.com/)

Every pull request will also show who approved. The only thing that is currently keeping me from switching it on is their [Privacy Policy](https://pullapprove.com/privacy/), where they mention third parties like GoogleAnalytics. Below is how `PullApprove` works explained in one short GIF. How do you think about adding this, @larsgrefer?

[![PullApprove at work](https://pullapprove.com/static/img/approve-by-comment.gif)](https://pullapprove.com/)

**Comment by agilob** [comment]
> Since we've too often had the case that stuff gets merged and later on some developers like @agilob complained that noone seemed to have reviewed it before the merge, 

Nothing changed since that time: #682 https://github.com/SecUpwN/Android-IMSI-Catcher-Detector/pull/697 https://github.com/SecUpwN/Android-IMSI-Catcher-Detector/pull/685 https://github.com/SecUpwN/Android-IMSI-Catcher-Detector/pull/683  https://github.com/SecUpwN/Android-IMSI-Catcher-Detector/pull/670 https://github.com/SecUpwN/Android-IMSI-Catcher-Detector/pull/669

**Comment by larsgrefer** [comment]
If a pull request is not merged by the person who created it, we have at least two reviewers.

@SecUpwN How many approves would you like to have per PR?

**Comment by SecUpwN** [comment]
>@SecUpwN How many approves would you like to have per PR?

@larsgrefer, I think two would be fine. since I know that you know what you do, I guess adding you as a reviewer works good. We need one more stable Android developer, which we still lack, unfortunately.

**Comment by agilob** [comment]
>If a pull request is not merged by the person who created it, we have at least two reviewers.

The PRs I linked had no reviews, some were merged by creator, introduce regressions/increase apk size without any good reason/break build/break functionality/slow down application/introduce vulnerabilities or contain shitty commit messages that should never land in /master...

**Comment by larsgrefer** [comment]
@agilob The PRs you linked were approved by @SecUpwN and me
How much has the apk size increased?
What functionality broke?
What vulnerabilities were introduced?

**Comment by agilob** [comment]
>@agilob The PRs you linked were approved by @SecUpwN and me

He's not a developer, so it doesn't count.

> How much has the apk size increased?

Nearly twice so far.

> What functionality broke?

Application doesn't start without GPS. Application doesn't start as system app. Is it still crashing when detecting SMS-0?

> What vulnerabilities were introduced?

Memory leaks, low quality dependencies without community/review and audits were added, Secs keystore will be published on github.

**Comment by larsgrefer** [comment]
@agilob Let me correct/clarify some points:
- The Application does start without GPS
- The Application can only run as system app if its signed with the same key as all other system apps
  - the system product-flavor is only for use by rom-creators
- No, the app isn't crashing anymore when a silent SMS is detected
- The keystore will be AES-256 encrypted

What memory leaks were introduced?

**Comment by SecUpwN** [comment]
>He's not a developer, so it doesn't count.

Main lessen I learned from all the time here on GitHub: Rants roll right off my back. :smile_cat: 

>If a pull request is not merged by the person who created it, we have at least two reviewers.

@larsgrefer, exactly. I chose you, then even if not always online @He3556 and @DJaeger. Looking at [our team](https://github.com/SecUpwN/Android-IMSI-Catcher-Detector/wiki/Contact), maybe even @tobykurien want to join the review process? @smarek would be lovely as well.
